src/test/java/test/listeners/github1465/ExampleClassListener.java

```diff
@@ -0,0 +1,63 @@
+package test.listeners.github1465;
+
+import org.testng.IInvokedMethod;
+import org.testng.IInvokedMethodListener;
+import org.testng.ITestNGMethod;
+import org.testng.ITestResult;
+
+import java.lang.reflect.Method;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ExampleClassListener implements IInvokedMethodListener {
+    List<String> messages = new LinkedList<>();
+    List<String> configMsgs = new LinkedList<>();
+
+    @Override
+    public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
+        log("beforeInvocation:", method, testResult);
+    }
+
+    @Override
+    public void afterInvocation(IInvokedMethod method, ITestResult testResult) {
+        log("afterInvocation:", method, testResult);
+    }
+
+    private void log(String prefix, IInvokedMethod method, ITestResult testResult) {
+        String msg = prefix + "_" + typeOfMethod(method);
+        msg += method.getTestMethod().getMethodName() + parameters(testResult);
+        if (method.isConfigurationMethod()) {
+            configMsgs.add(msg);
+        } else {
+            messages.add(msg);
+        }
+    }
+
+    private static String typeOfMethod(IInvokedMethod method) {
+        ITestNGMethod tm = method.getTestMethod();
+        if (tm.isBeforeMethodConfiguration()) {
+            return "before_method: ";
+        }
+        if (tm.isAfterMethodConfiguration()) {
+            return "after_method: ";
+        }
+        return "test_method: ";
+    }
+
+    private static String parameters(ITestResult testResult) {
+        Object[] parameters = testResult.getParameters();
+        if (parameters != null) {
```

Comment:
I think quick return helps to highlight the important code vs check code:

```java
if (parameters == null) {
  return "";
}

// ...

Reply:
Was trying to avoid having to introduce multiple exits in this method, especially when its branching construct is pretty small and fits a page when viewed in an IDE. Anyways.. will change it.

Reply:
Again, true. It was clearly debatable here.